File: solution/array/151-reverse-words-in-a-string/solution.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
    const words = s.trim().split(/\s+/);

    let left = 0;
    let right = words.length - 1;

    while (left < right) {
        [words[left], words[right]] = [words[right], words[left]];
        left++;
        right--;
    }

    return words.join(" ");
};

File: solution/array/151-reverse-words-in-a-string/solution.go
package main

import (
	"strings"
)

func reverseWords(s string) string {
	words := strings.Fields(s)

	for i := 0; i < len(words)/2; i++ {
		j := len(words) - 1 - i
		words[i], words[j] = words[j], words[i]
	}

	return strings.Join(words, " ")
}